pubsublite/admin/pubsublite_test.go

```diff
@@ -0,0 +1,480 @@
+	sub        = "test-sub-"
+	testRegion = "us-central1"
+)
+
+var (
+	supportedZoneIDs = []string{"a", "b", "c"}
+
+	once       sync.Once
+	projNumber string
+)
+
+func setupAdmin(t *testing.T) *pubsublite.AdminClient {
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
+
+	client, err := pubsublite.NewAdminClient(ctx, testRegion)
+	if err != nil {
+		t.Fatalf("failed to create client: %v", err)
+	}
+
+	once.Do(func() {
+		rand.Seed(time.Now().UnixNano())
+		// PubSub Lite returns project numbers in resource paths, so we need to convert from project id
+		// to numbers for tests.
+		cloudresourcemanagerService, err := cloudresourcemanager.NewService(context.Background())
+		if err != nil {
+			t.Fatalf("cloudresourcemanager.NewService: %v", err)
+		}
+
+		project, err := cloudresourcemanagerService.Projects.Get(tc.ProjectID).Do()
+		if err != nil {
+			t.Fatalf("cloudresourcemanagerService.Projets.Get project: %v", err)
+		}
+
+		projNumber = strconv.FormatInt(project.ProjectNumber, 10)
+
+		cleanup(t, client, projNumber)
+	})
+
+	return client
+}
+
+// cleanup deletes all previous test topics/subscriptions from
+// previous test runs. This prevents previous test failures
+// from building up resources that count against quota.
+func cleanup(t *testing.T, client *pubsublite.AdminClient, proj string) {
+	ctx := context.Background()
+
+	for _, zoneID := range supportedZoneIDs {
+		zone := fmt.Sprintf("%s-%s", testRegion, zoneID)
+		parent := fmt.Sprintf("projects/%s/locations/%s", proj, zone)
+		topicIter := client.Topics(ctx, parent)
+		for {
+			topic, err := topicIter.Next()
+			if err == iterator.Done {
+				break
+			}
+			if err != nil {
+				t.Fatalf("topicIter.Next got err: %v", err)
+			}
+			if err := client.DeleteTopic(ctx, topic.Name); err != nil {
+				t.Fatalf("client.DeleteTopic got err: %v", err)
+			}
+		}
+
+		subIter := client.Subscriptions(ctx, parent)
+		for {
+			sub, err := subIter.Next()
+			if err == iterator.Done {
+				break
+			}
+			if err != nil {
+				t.Fatalf("subIter.Next() got err: %v", err)
+			}
+			if err := client.DeleteSubscription(ctx, sub.Name); err != nil {
+				t.Fatalf("client.DeleteSubscription got err: %v", err)
+			}
+		}
+	}
+}
+
+func TestCreateTopic(t *testing.T) {
+	client := setupAdmin(t)
+	tc := testutil.SystemTest(t)
+	testZone := randomZone()
+
+	topicID := topic + uuid.NewString()
+	topicPath := fmt.Sprintf("projects/%s/locations/%s/topics/%s", projNumber, testZone, topicID)
+	buf := new(bytes.Buffer)
+	err := createTopic(buf, tc.ProjectID, testRegion, testZone, topicID)
+	if err != nil {
+		t.Fatalf("createTopic: %v", err)
+	}
+	got := buf.String()
+	want := fmt.Sprintf("Created topic: %s\n", topicPath)
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Fatalf("createTopic() mismatch: -want, +got:\n%s", diff)
+	}
+
+	t.Cleanup(func() {
+		client.DeleteTopic(context.Background(), topicPath)
+	})
+}
+
+func TestGetTopic(t *testing.T) {
+	client := setupAdmin(t)
```

Comment:
Should we defer client.Close() for all tests?
I've found that it leaves open connections and goroutines. But only experienced any issues if I run tests with `-count` many times.